src/pages/logout.js

```diff
@@ -0,0 +1,15 @@
+import { useHistory } from "react-router";
+import { useEffect } from "react";
+import { AUTH } from "../components/constants";
+import { logout } from "../services/userService";
+
+function LogOut(props) {
+  const history = useHistory();
+  useEffect(() => {
+    logout();
+    history.replace(AUTH);
+    props.setLoggedIn(false);
+  }, [history, props]);
```

Comment:
`[history, props]` in `useEffect` means callback will be called every time `history` or `props` change. We don't want to do this we just want this callback to be invoked only once when the component mounts, to do so we pass an empty dependency array to `useEffect`.
```suggestion
  }, []);
```